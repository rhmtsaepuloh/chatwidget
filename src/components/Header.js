import React from "react";
import { useSelector } from "react-redux";
import { FiArrowLeft } from "react-icons/fi";

import "../styles/Header.scss";
import isEmpty from "lodash/isEmpty";
import { defaultText } from "../utils/ui";
import { truncateWords } from "../utils/messages";

const avatarIcon = require("../assets/logo.jpg");

const Header = React.memo(({ onGoBack }) => {
  const { primaryColor, primaryTextColor, logo } = useSelector(
    (state) => state.clientReducers.dataClient
  );
  const { contentRoomChat } = useSelector(
    (state) => state.clientReducers.content
  );
  return (
    <div
      className='header flex items-center'
      style={{ background: primaryColor, overflow: "hidden" }}>
      <FiArrowLeft
        className='arrow-icon text-white text-base font-bold cursor-pointer text-2xl'
        onClick={onGoBack}
      />
      <img
        alt='icon'
        className='avatarIcon'
        src={isEmpty(logo) ? avatarIcon : logo}
      />
      <div>
        <p className='text-koperansel' style={{ color: primaryTextColor }}>
          {truncateWords(
            defaultText(contentRoomChat, "nameTitle", "@ChatAja"),
            3,
            "..."
          )}
        </p>
        <p className='text-online' style={{ color: primaryTextColor }}>
          online
        </p>
      </div>
    </div>
  );
});

export default Header;
