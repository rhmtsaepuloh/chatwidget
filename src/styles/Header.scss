.header {
  padding: 20px;
  box-sizing: border-box;
  position: fixed;
  overflow: hidden;
  top: 0;
  height: 90px;
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
  @media (max-width: 768px) {
    width: 100%;
  }
  @media (min-width: 768px) {
    width: 378px;
  }
  .avatarIcon {
    height: 30px;
    width: 30px;
    border-radius: 20px;
    margin: 0px 8px;
  }
  div {
    // @media (max-width: 378px) {
    //   width: 50%;
    // }
    .text-koperansel {
      color: #fff;
      font-size: 15px;
      font-weight: bold;
      white-space: nowrap;
      width: 95%;
    }
    .text-online {
      color: #fff;
      font-size: 14px;
    }
  }
}
