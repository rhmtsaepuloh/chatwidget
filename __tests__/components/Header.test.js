/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";
import { useSelector } from "react-redux";

import Header from "../../src/components/Header";

const onGoBack = jest.fn();

jest.mock("react-redux", () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

describe("render correctly header components", () => {
  test("renders with contain data selector", () => {
    useSelector.mockImplementation((selector) =>
      selector({
        clientReducers: {
          dataClient: {
            name: "Chat Aja",
            hexColor: "#CA1234",
            logo:
              "https://chatwidget-chataja.s3.ap-southeast-1.amazonaws.com/0fcedfb7-1861-42c0-9515-13603ce62901.png",
          },
          content: {
            contentRoomChat: {
              nameTitle: "@ChatAja",
            },
          },
        },
      })
    );
    const wrapper = shallow(<Header onGoBack={onGoBack} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders without contain data selector", () => {
    useSelector.mockImplementation((selector) =>
      selector({
        clientReducers: {
          dataClient: {
            name: "Chat Aja",
            hexColor: "",
            logo: "",
          },
          content: {
            contentRoomChat: {
              nameTitle: "@ChatAja",
            },
          },
        },
      })
    );
    const wrapper = shallow(<Header onGoBack={onGoBack} />);
    expect(wrapper).toMatchSnapshot();
  });
});
