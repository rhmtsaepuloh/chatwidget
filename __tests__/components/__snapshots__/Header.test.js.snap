// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly header components renders with contain data selector 1`] = `ShallowWrapper {}`;

exports[`render correctly header components renders without contain data selector 1`] = `ShallowWrapper {}`;
